### PR TITLE
Add mod_lbmethod_cluster.conf.

### PR DESCRIPTION
**Pull request by jfclere**
And add:
Some docs for it.
Arrange DEBUG and some problem when building images.

### CODE REVIEW

File: test/includes/common.sh [hunk 1]
```diff
@@ -41,6 +41,16 @@ run_test() {
 httpd_create() {
     rm -rf httpd/mod_proxy_cluster /tmp/mod_proxy_cluster
     mkdir /tmp/mod_proxy_cluster
+    # make sure the native are cleaned before copying
+    for m in advertise mod_proxy_cluster balancers mod_manager
+    do
+        cd $m
```
Comment (jajik) [inline test/includes/common.sh:47]: There is `native` missing, it should be `cd ../native/$m`.

File: test/testsuite.sh [hunk 2]
```diff
@@ -28,15 +28,28 @@ echo "        TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT"
...
     mkdir logs
 fi
 
 . includes/common.sh
 
-httpd_create  > /dev/null 2>&1 || exit 2
-tomcat_create > /dev/null 2>&1 || exit 3
+if [ -d target ]; then
+  echo "please run setup-dependencies.sh before!!!"
+  exit 1
```
Comment (jajik) [inline test/testsuite.sh:43]: `exit 4` would be better; it's better to have different exit code for each situation (see the beginning of the file).

File: test/testsuite.sh [hunk 3]
```diff
@@ -28,15 +28,28 @@ echo "        TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT"
...
 if [ ! -d logs ]; then
     mkdir logs
 fi
 
 . includes/common.sh
 
-httpd_create  > /dev/null 2>&1 || exit 2
-tomcat_create > /dev/null 2>&1 || exit 3
+if [ -d target ]; then
+  echo "please run setup-dependencies.sh before!!!"
```
Comment (jajik) [inline test/testsuite.sh:42]: `"Missing dependencies. Please run setup-dependencies.sh and then try again"`, WDYT?

File: test/testsuite.sh [hunk 4]
```diff
@@ -28,15 +28,28 @@ echo "        TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT"
 echo "        ITERATION_COUNT=$ITERATION_COUNT"
 echo "        IMG=$IMG"
 echo "        HTTPD_IMG=$HTTPD_IMG"
+if [ ! -z ${HTTPD_IMG+x} ]; then
```
Comment (jajik) [inline test/testsuite.sh:31]: You meant `! -z ${HTTPD_IMG+x}` probably..?

File: test/README.md [hunk 5]
```diff
@@ -75,3 +75,8 @@ There is also a python script that can be run to check mod_proxy_cluster. You ca
 as `miniserver.py`. Execute it simply as `./includes/miniserver.py <port-number>`.
 
 You must have `httpd` with `mod_proxy_cluster` running.
+
+# Testing the lbmethod and mod_proxy_balancer
+Just run the test with CONF=httpd/mod_lbmethod_cluster.conf
+```
```
Comment (jajik) [inline test/README.md:81]: Missing closing ```.

File: test/includes/common.sh (hunk 1 above)
Comment (jfclere) [reply test/includes/common.sh:47]: thanks fixed.

File: test/testsuite.sh (hunk 2 above)
Comment (jfclere) [reply test/testsuite.sh:43]: fixed.

File: test/testsuite.sh (hunk 4 above)
Comment (jfclere) [reply test/testsuite.sh:31]: Oops I meant MPC_CONF

Comment (jajik) [reply test/testsuite.sh]: Yeah, me too :D

File: test/README.md (hunk 5 above)
Comment (jajik) [reply test/README.md:81]: This is still relevant (it displays properly only because its at the end of file)-

**Review by jajik — CHANGES_REQUESTED** [review]
Two things:

* Is the whole testsuite relevant for testing with lbmethod + balancer? If no, I would suggest picking only tests, that are relevant. In both cases, I think it would be good to add test cases directly to the testsuite instead of forcing user to run the testsuite twice.
* It would be nicer to squash commits (there is no need to merge fixed separatedly).

File: test/README.md (hunk 5 above)
Comment (jfclere) [reply test/README.md:81]: https://github.com/jfclere/mod_proxy_cluster/blob/master/test/README.md?plain=1#L83
It is there, what do I miss?

Comment (jajik) [reply test/README.md:81]: WTH, I must be blind! I'm sorry, you're right.

**Comment by jajik** [comment]
> > * Is the whole testsuite relevant for testing with lbmethod + balancer?
> 
> Tests that wait for all workers to be removed might be failing but most of the tests should pass.

Ok, that looks like what I'm seeing locally. Let's merge this after applying those two patches above, and I will then incorporate it into the testsuite (so that it is run in CI and the user does not have to run it twice manually).

**Comment by jfclere** [comment]
Actually with mod_lbmethod_cluster.conf there are many:
[Thu Oct 05 09:54:04.191970 2023] [proxy_balancer:error] [pid 12:tid 140571226126016] [client 127.0.0.1:46670] AH10164: balancer://mycluster: failed to add worker ws://127.0.0.1:8080
Probably we need to check if the worker already exits before adding it in this case, no?

**Comment by jfclere** [comment]
I have added the 2 patches I think we are good to merge now ;-)

**Comment by jajik** [comment]
> Actually with mod_lbmethod_cluster.conf there are many: [Thu Oct 05 09:54:04.191970 2023] [proxy_balancer:error] [pid 12:tid 140571226126016] [client 127.0.0.1:46670] AH10164: balancer://mycluster: failed to add worker ws://127.0.0.1:8080 Probably we need to check if the worker already exits before adding it in this case, no?

I briefly looked at it, and it's weird. I'm going to investigate it further once we merge this so we won't delay this PR.

**Comment by jajik** [comment]
> I have added the 2 patches I think we are good to merge now ;-)

I thought you would apply them separately as they are independent of each other. Could you at least change the commit message so it is clear that the assert is removed and logs tweaked? Also, if you removed the "submitted by jajik" part, it would be nice (it's not relevant for the commit message although I truly appreciate it [^1])

[^1]: If you really wish to mention me as a co-author, there is apparently [a way](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) how to do it supported on both, GitHub and Gitlab: just add `Co-authored-by: vchlup@redhat.com` on a separate line in the commit message).

File: test/testsuite.sh [hunk 6]
```diff
@@ -28,15 +28,28 @@ echo "        TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT"
 echo "        ITERATION_COUNT=$ITERATION_COUNT"
 echo "        IMG=$IMG"
 echo "        HTTPD_IMG=$HTTPD_IMG"
+if [ ! -z ${MPC_CONF+x} ]; then
+echo "        MPC_CONF=$MPC_CONF"
```
Comment (jajik) [inline test/testsuite.sh:32]: I don't want to be nitpicking, but the echo would be better indented.

File: test/testsuite.sh [hunk 7]
```diff
@@ -28,15 +28,28 @@ echo "        TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT"
...
 
 if [ ! -d logs ]; then
     mkdir logs
 fi
 
 . includes/common.sh
 
-httpd_create  > /dev/null 2>&1 || exit 2
-tomcat_create > /dev/null 2>&1 || exit 3
+if [ -d target ]; then
```
Comment (jajik) [inline test/testsuite.sh:41]: Shouldn't there be a `!`?

Comment (jajik) [reply test/testsuite.sh:41]: Trying it on a fresh repository, it should be `! -d tomcat/target` actually.

Comment (jfclere) [reply test/testsuite.sh:41]: fixed

File: test/testsuite.sh (hunk 6 above)
Comment (jfclere) [reply test/testsuite.sh:32]: fixed.

**Review by jajik — APPROVED** [review]
LGTM, thanks for the patience :)

**Pull request merged**